flowmachine/flowmachine/features/utilities/event_table_subset.py

```diff
@@ -164,7 +173,47 @@ def _check_dates(self):
                 stacklevel=2,
             )
 
-    def _make_query(self):
+    def _make_query_with_sqlalchemy(self):
+        table = get_sqlalchemy_table_definition(
+            self.table.fully_qualified_table_name, engine=Query.connection.engine
+        )
+        sqlalchemy_columns = [
+            get_sqlalchemy_column(table, column_str) for column_str in self.columns
+        ]
+        select_stmt = select(sqlalchemy_columns)
+
+        if self.start is not None:
+            ts_start = pd.Timestamp(self.start).strftime("%Y-%m-%d %H:%M:%S")
+            select_stmt = select_stmt.where(table.c.datetime >= ts_start)
+        if self.stop is not None:
+            ts_stop = pd.Timestamp(self.stop).strftime("%Y-%m-%d %H:%M:%S")
+            select_stmt = select_stmt.where(table.c.datetime <= ts_stop)
+
+        if self.hours != "all":
+            hour_start, hour_end = self.hours
+            if hour_start < hour_end:
+                select_stmt = select_stmt.where(
+                    between(extract("hour", table.c.datetime), hour_start, hour_end - 1)
+                )
+            else:
+                # If dates are backwards, then this will be interpreted as spanning midnight
+                select_stmt = select_stmt.where(
+                    or_(
+                        extract("hour", table.c.datetime) >= hour_start,
+                        extract("hour", table.c.datetime) < hour_end,
+                    )
+                )
+
+        select_stmt = self.subscriber_subset.apply_subset(
```

Comment:
See my comment above. I would consider using an if-statement and only calling when we want to subset by a subscriber.

Reply:
See reply above, I'd very much like to not do this because it's one of the main sources of coupling in the code base at the moment.